doc/tutorials/contributing-clion.md

```diff
@@ -0,0 +1,183 @@
+and type:
+
+```sh
+git clone https://github.com/ElektraInitiative/libelektra.git
+```
+
+With the project now locally available we can start developing.
+
+## Setting up the project
+
+It is important to start CLion as root-user, since some functionalities of
+_kdb_ require superuser privileges to work as expected. Once you see the
+main menu, click "Open" and select the CMakeLists.txt-file inside the
+project's root directory. This will import the project accordingly and
+populate you run-configuration with some predefined values. In rare
+occurrences this won't happen. If that is the case for you, simply restart
+CLion using:
+
+_File_ --> _Invalidate Caches / Restart..._ --> _Invalidate and Restart_
+
+Now after all processes of CLion have finished, the project should be set up
+and the run-configuration should be populated with entries.
+
+## Development
+
+Usually the folders you have to work in to add functionality or
+documentation are as follows:
+
+- doc<br/>
+  This folder contains mainly all documentation of the project, including
+  almost all pages of the [homepage](https://www.libelektra.org) of this
+  project. One important note is, that all markdown-pages can also be used
+  for testing using
+  [Markdown to Shell Recorder](https://github.com/ElektraInitiative/libelektra/tree/master/tests/shell/shell_recorder/tutorial_wrapper)
+  (you can find an example on how to do this [here](/doc/help/kdb-get.md)).
+- src<br/>
+  Almost all functionality-code resides here.
+  - bindings<br/>
+    Here is all the code of available [Bindings](/src/bindings/README.md)
+    of libelektra. - plugins<br/>
+    You can find all developed [Plugins](/src/plugins/README.md) here. If
+    you want to fix a bug for an existing plugin or add another one, this is
+    the directory you have to work in. For further information on how to
+    develop your own plugin, please visit [this](/doc/tutorials/plugins.md)
+    tutorial. - tools<br/>
+    In this folder the source of all tools for interacting with Electra's
+    global key database, e.g. _kdb_, can be found.
+- tests<br/>
+  Here you can find all sorts of tests (excluding tests created with the
+  _Markdown to Shell Recorder_-tool).
+
+## Testing
+
+The most thorough way to test your changes is to run all tests. Therefore
+navigate to your run-configurations (Run -> Edit Configurations...) and look
+for the entry "run*all". There, "run_all" should be selected as \_Executable*,
+"kdb" as _Target_. Now you can execute this run-configuration which will run
+all enabled tests!
+
+If you want to test various _kdb_-methods separately, you can create your
+own run-configurations. Add a new one by clicking on the "+"-sign on the
+top left of the "Edit Configurations..."-window and name it. Here both
+_Executable_ and _Target_ should have "kdb" selected. If you for example
+want to test `kdb info dump`, write "info dump" next to _Program arguments_.
+That's it, now you can just test this part of _kdb_.
```

Comment:
You can also use a run configuration with executable and target set to any of the `testmod_*` or `testkdb_*` targets to run these tests.

Additionally for any of the tests that use Google Test, Clion's integrated Test runner may be used. The easiest way to create those run configurations is to just open the test source and click the icon next to the class.

For the `testscr_*` and `testshell_*` tests the best way of running is via the integrated terminal and `ctest` (after ensuring the correct parts of Elektra are built).